coderd/files/cache_internal_test.go

```diff
@@ -0,0 +1,102 @@
+package files
+
+import (
+	"context"
+	"io/fs"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/coder/coder/v2/testutil"
+)
+
+var emptyFS fs.FS = afero.NewIOFS(afero.NewMemMapFs())
```

Comment:
Just put this in each test, let's avoid doing any globals.

And can we make it readonly? : `afero.NewIOFS(afero.NewReadOnlyFs(afero.NewMemMapFs()))`